--- dyninst-7.99/dyninst/make.config.git	2012-09-18 10:56:40.000000000 -0700
+++ dyninst-7.99/dyninst/make.config	2012-09-18 15:14:40.387837651 -0700
@@ -5,7 +5,7 @@
 #
 
 SUITE_NAME	= Paradyn
-RELEASE_NUM	= 8.0
+RELEASE_NUM	= 7.99.2
 
 ifndef TO_CORE
 TO_CORE		= ../..
--- dyninst-7.99/dyninst/dyninstAPI/h/BPatch.h.git	2012-09-18 15:13:36.748990056 -0700
+++ dyninst-7.99/dyninst/dyninstAPI/h/BPatch.h	2012-09-18 15:14:06.147920976 -0700
@@ -61,8 +61,8 @@ class func_instance;
 #define DYNINST_7_0
 
 #define DYNINST_MAJOR 7
-#define DYNINST_MINOR 0
-#define DYNINST_SUBMINOR 0
+#define DYNINST_MINOR 99
+#define DYNINST_SUBMINOR 2
 
 #ifdef IBM_BPATCH_COMPAT
 typedef void *BPatch_Address;
