
--- ./dyninst-10.0.0/CMakeLists.txt.sv	2018-11-09 11:48:08.000000000 -0500
+++ ./dyninst-10.0.0/CMakeLists.txt	2018-11-13 11:35:20.109650373 -0500
@@ -149,7 +149,7 @@ if(BUILD_DOCS)
   foreach(COMPONENT ${WORD_DOCS})
     add_dependencies(doc ${COMPONENT}-doc)
     set_target_properties(${COMPONENT}-doc PROPERTIES EXCLUDE_FROM_DEFAULT_BUILD 1)
-    install(FILES ${CMAKE_SOURCE_DIR}/${COMPONENT}/doc/${COMPONENT}.pdf
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${COMPONENT}/doc/${COMPONENT}.pdf
       DESTINATION ${INSTALL_DOC_DIR}
       RENAME ${COMPONENT}-${VERSION_STRING}.pdf
       OPTIONAL
