diff -up dyninst-7.99/dyninst/symtabAPI/src/relocationEntry-elf-ppc32.C.ppc dyninst-7.99/dyninst/symtabAPI/src/relocationEntry-elf-ppc32.C
--- dyninst-7.99/dyninst/symtabAPI/src/relocationEntry-elf-ppc32.C.ppc	2012-08-15 07:50:54.270065052 -0400
+++ dyninst-7.99/dyninst/symtabAPI/src/relocationEntry-elf-ppc32.C	2012-08-15 07:50:10.500111439 -0400
@@ -101,7 +101,9 @@ const char* relocationEntry::relType2Str
         CASE_RETURN_STR(R_PPC_GOT_DTPREL16_LO);
         CASE_RETURN_STR(R_PPC_GOT_DTPREL16_HI);
         CASE_RETURN_STR(R_PPC_GOT_DTPREL16_HA);
+#if defined(R_PPC_NUM)
         CASE_RETURN_STR(R_PPC_NUM);
+#endif
         CASE_RETURN_STR(R_PPC_EMB_NADDR32);
         CASE_RETURN_STR(R_PPC_EMB_NADDR16);
         CASE_RETURN_STR(R_PPC_EMB_NADDR16_LO);
