diff --git a/common/i386-unknown-linux2.4/Makefile b/common/i386-unknown-linux2.4/Makefile
index d92aedc..9663864 100644
--- a/common/i386-unknown-linux2.4/Makefile
+++ b/common/i386-unknown-linux2.4/Makefile
@@ -4,14 +4,8 @@
 
 # Define any symbols needed to invoke configuration changes in make.config
 
-#USES_SHM_SAMPLING = true
-#USES_PAPI = true
-DEBUG_FLAG = -g
 NO_IMPLICIT_TEMPLATES = true
 
-# For debugging.
-#NO_OPT_FLAG = true
-
 # Include standard make configuration stuff that applies to everything
 # in the paradyn tree.
 
@@ -24,35 +18,18 @@ SRCS =		../../common/src/linuxKludges.C \
 		../../common/src/parseauxv.C \
 		../../common/src/addrtranslate-sysv.C \
 		../../common/src/addrtranslate-auxv.C \
-		../../common/src/addrtranslate-linux.C \
-		../../common/src/dwarfExpr.C
-
-#MYFLAGS	+= -DNO_MDL_PRINT
-#MYFLAGS	+= -DLD_DEBUG
-#MYFLAGS	+= -DATTACH_DETACH_DEBUG
-#MYFLAGS	+= -DPTRACEDEBUG
-#MYFLAGS	+= -DPTRACEDEBUG_EXCESSIVE
-#MYFLAGS	+= -DPTRACEDEBUG_ALWAYS
-#MYFLAGS	+= -DINST_TRAP_DEBUG
-#MYFLAGS	+= -DSIGNAL_DEBUG
-#MYFLAGS	+= -DASTDEBUG
-#MYFLAGS	+= -DINFERIOR_RPC_DEBUG
+		../../common/src/addrtranslate-linux.C
 
 CXXFLAGS        += $(MYFLAGS)
 CFLAGS          += $(MYFLAGS)
 TFLAGS          += $(MYFLAGS)
 
-# Tradeoff: adding -g to TFLAGS aids gdb but uses gobs of disk space
-#TFLAGS		+= -g
-
 LIBS    += -ldl
 
 AS		= /usr/bin/as
 
-
 # Add preprocessor and special link information for G++
-G_PTHREAD	= -pthread
-G_PTHREAD_LD	= -pthread
+G_PTHREAD       = -pthread
 
 # Include the module-specific Makefile, which defines everything about
 # the module that is common across architectures.
